corehq/motech/forms.py

```diff
@@ -168,7 +172,7 @@ def helper(self):
                     crispy.Field('pass_credentials_in_header'),
                     crispy.Field('include_client_id'),
                     crispy.Field('scope'),
-                    crispy.Field('plaintext_custom_headers'),
+                    crispy.Field('plaintext_custom_headers') if custom_headers_supported else None,
```

Comment:
nit: does crispy form accept `None` well for fields?
Usually I have seen in code fields getting added only when needed or else nothing added. So, I would think this would be
(would need tweaking fixes)
```
def _oauth_fields():
    fields = [
        _("OAuth 2.0 Settings"),
        crispy.Field('auth_preset'),
        # ...
         crispy.Field('scope'),
    ]
    if custom_headers_supported:
        fields.append(crispy.Field('plaintext_custom_headers'))
    return fields
```
and then update this to 
```
crispy.Fieldset(**self.oauth_fields())
```

Reply:
Also, just realized, we are adding custom headers only for OAuth?
If its not too much of a change, can we make it available for any request type?

Reply:
> does crispy form accept None well for fields?

Yes, seems to work just fine.

> we are adding custom headers only for OAuth?

Correct. I can create a ticket for adding it to the other auth types as well, but for now this is to unblock [this work](https://github.com/dimagi/commcare-hq/pull/35966).

Reply:
thanks @Charl1996 , sounds good to take that later.